docs: update README.md to enhance documentation for Docker image features and usage

vOrigins opened the pull request:
- Added project description and features overview
- Included instructions for screenshot testing with Jest and Puppeteer
- Provided CI/CD integration examples and direct usage instructions
- Updated installed software section with version details

vOrigins committed:
docs: update README.md to enhance documentation for Docker image features and usage

- Added project description and features overview
- Included instructions for screenshot testing with Jest and Puppeteer
- Provided CI/CD integration examples and direct usage instructions
- Updated installed software section with version details

JayLim2 commented on README.md:
```diff
@@ -1,12 +1,113 @@
-Development Environment
+![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/netcracker/qubership-apihub-nodejs-dev-image/docker-ci.yml?label=build)
+
+A development environment Docker image specifically designed for screenshot testing of NodeJS applications with Chrome and Puppeteer support. Used as a base image for development, testing, and CI/CD pipelines in the qubership-apihub ecosystem.
```
"NodeJS application" means backend service written on NodeJS.
I suggest to write "JS-written frontend applications" or something like that.

b41ex replied:
let's just use 'frontend applications'

vOrigins replied:
fixed

b41ex commented on README.md:
```diff
@@ -1,12 +1,113 @@
+
+### Using with Jest and Puppeteer
+
+To use this image for screenshot testing with Jest and Puppeteer, you can leverage the `@netcracker/qubership-apihub-jest-chrome-in-docker-environment` package:
```
Let's add a link to repository here

vOrigins replied:
fixed

b41ex commented on README.md:
````diff
@@ -1,12 +1,113 @@
+)
+```
+
+This configuration runs your Jest tests inside the Docker container, ensuring a consistent environment for your screenshot tests.
````
Let's remove 'ensuring a consistent environment for your screenshot tests', it just repeats almost the same phrase above.

vOrigins replied:
fixed

b41ex commented on README.md:
```diff
@@ -1,12 +1,113 @@
+
+## CI/CD Integration
+
+This image is used in CI/CD pipelines for building and testing JavaScript/TypeScript applications. Here's an example of using it in GitHub Actions:
```
I believe, this is assuming `npm test` contains some screenshot tests? Let's clarify this

vOrigins replied:
replaced the example with a link with a real use case in CI/CD

b41ex commented on README.md:
````diff
@@ -1,12 +1,113 @@
+      - run: npm test
+```
+
+## Direct Usage
````
What is the use case for direct usage?

vOrigins replied:
removed direct usage as redundant

vOrigins committed:
docs: update README to improve clarity and add relevant links

- Revised project description to emphasize frontend application support
- Enhanced instructions for using the image with Jest and Puppeteer
- Updated CI/CD integration section with a link to the workflow file
- Added local build instructions for the Docker image
- Removed Direct Usage, License and Security sections as redundant

b41ex committed:
docs: minor modification
